src/Compilers/Core/MSBuildTask/CopyRefAssembly.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    /// <summary>
+    /// If the source and destination ref assemblies are different,
+    /// this task copies the source over to the destination.
+    /// </summary>
+    public class CopyRefAssembly : Task
+    {
+        [Required]
+        public string SourcePath { get; set; }
+
+        [Output]
+        [Required]
+        public string DestinationPath { get; set; }
+
+        public CopyRefAssembly()
+        {
+            TaskResources = ErrorString.ResourceManager;
+        }
+
+        public override bool Execute()
+        {
+            if (!File.Exists(SourcePath))
+            {
+                Log.LogErrorWithCodeFromResources("General_ExpectedFileMissing", SourcePath);
+                return false;
+            }
+
+            if (!File.Exists(DestinationPath))
```

Comment:
Making us redundant to extractMvid failures would also allow you to remove this check.  Or at least centralize it in ExtractMvid.